### PR TITLE
[Keyboard] matchstickworks/southpad - Move files to rev1, add rev2

### PR DESCRIPTION
**Pull request by MatchstickWorks**
## Description

- Moved the original files to rev1 folder with changes only to readme.md to differentiate between rev1 and rev2.
- Adding rev2 board of the same name/layout with different MCU and adding via support. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by MatchstickWorks** [comment]
Working through QMK CI Build errors. Fixing shortly.

File: keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
+		KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
+        KC_PMNS, KC_PAST, KC_PSLS, TG(1),
+        KC_PPLS, KC_P9,   KC_P8,   KC_P7,
+        KC_P6,   KC_P5,   KC_P4,
+        KC_PENT, KC_P3,   KC_P2,   KC_P1,
+        KC_PDOT, KC_P0
+	),
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c:15]: Ideally, this should be rearranged to more closely match the normal numpad layout. 
```suggestion
	[0] = LAYOUT(
		KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
        KC_PMNS, KC_PAST, KC_PSLS, TG(1),
                 KC_P9,   KC_P8,   KC_P7,
        KC_PPLS, KC_P6,   KC_P5,   KC_P4,
                 KC_P3,   KC_P2,   KC_P1,
        KC_PENT, KC_PDOT,      KC_P0
	),
```

eg: 
https://github.com/qmk/qmk_firmware/blob/master/layouts/default/numpad_6x4/default_numpad_6x4/keymap.c

While I know it's mirrored, consistency would be ideal.

File: keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	[3] = LAYOUT(
+		_______, _______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______
+	),
+
+};
```
Comment (dunk2k) [inline keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c:44]: non-functional layers
```suggestion
};

```

File: keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+	[3] = LAYOUT(
+		_______, _______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______, _______,
+		_______, _______, _______, _______,
+		_______, _______
+	),
+
```
Comment (dunk2k) [inline keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c:43]: Non-functional layers
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
@MatchstickWorks 
Would you be able to move respective `"maintainer"` and `"manufacturer"` entries from `/matchstickworks/southpaw/rev1/info.json` and `/matchstickworks/southpaw/rev2/info.json` into (newfile) `/matchstickworks/southpaw/info.json`

File: keyboards/matchstickworks/southpad/rev2/info.json
Comment (dunk2k) [inline keyboards/matchstickworks/southpad/rev2/info.json:1]: `layouts.LAYOUT.layout` doesn't match image in `readme.md` nor layout in `rev1/info.json`, please could you rectify?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c (hunk 1 above)
Comment (MatchstickWorks) [reply keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c:15]: I've adjusted the formatting to tab lines 12 and 14 over, but the pcd as is wired doesn't allow moving KC_PENT and KC_PPLS down to those lines. The formatting will be implemented in next commit.

**Comment by MatchstickWorks** [comment]
> @MatchstickWorks Would you be able to
> 
> 1. move respective `"maintainer"` and `"manufacturer"` entries from `/matchstickworks/southpaw/rev1/info.json` and `/matchstickworks/southpaw/rev2/info.json` into (new file) `/matchstickworks/southpaw/info.json`
> 2. migrate contents of `/matchstickworks/southpaw/rev1/rules.mk` to `/matchstickworks/southpaw/rev1/info.json`

This will be included in the next commit.

File: keyboards/matchstickworks/southpad/rev2/info.json
Comment (MatchstickWorks) [reply keyboards/matchstickworks/southpad/rev2/info.json:1]: I've updated the images in `readme.md`. Rev1 had the actual NumLock key, but it messes with how the arrows work, so rev2 uses a fake "NumLock" as a Toggle Layer key to properly define the arrows as intended. Is this something that works for you?

File: keyboards/matchstickworks/southpad/rev1/info.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "keyboard_name": "SouthPad",
+    "processor": "atmega32u2",
+    "bootloader": "atmel-dfu",
+    "usb": {
+        "vid": "0xFEED",
+        "pid": "0x6060",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "bootmagic": false,
```
Comment (waffle87) [inline keyboards/matchstickworks/southpad/rev1/info.json:11]: Ideally, this is left enabled if possible.

File: keyboards/matchstickworks/southpad/rev1/info.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        "mousekey": true,
+        "nkro": false
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["B0", "B1", "B2", "B4"],
+        "rows": ["B3", "B5", "B6", "B7", "C7", "C6"]
+    },
+    "layouts": {
+        "LAYOUT_numpad_6x4_southpaw": {
```
Comment (dunk2k) [inline keyboards/matchstickworks/southpad/rev1/info.json:24]:
```suggestion
        "LAYOUT": {
```
Would you be able to also apply this change to `rev1/keymaps/default/keymap.c`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/southpad/rev2/info.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                { "label": "/", "matrix": [1, 2], "x": 2.0, "y": 1.0 },
+                { "label": "Num Lock", "matrix": [1, 3], "x": 3.0, "y": 1.0 },
+                { "h": 2, "label": "+", "matrix": [2, 0], "x": 0.0, "y": 2.0 },
+                { "label": "9", "matrix": [2, 1], "x": 1.0, "y": 2.0 },
+                { "label": "8", "matrix": [2, 2], "x": 2.0, "y": 2.0 },
+                { "label": "7", "matrix": [2, 3], "x": 3.0, "y": 2.0 },
+                { "label": "6", "matrix": [3, 1], "x": 1.0, "y": 3.0 },
+                { "label": "5", "matrix": [3, 2], "x": 2.0, "y": 3.0 },
+                { "label": "4", "matrix": [3, 3], "x": 3.0, "y": 3.0 },
+                { "h": 2, "label": "Enter", "matrix": [4, 0], "x": 0.0, "y": 4.0 },
```
Comment (dunk2k) [inline keyboards/matchstickworks/southpad/rev2/info.json:41]: human readable formatting
```suggestion
                { "label": "+", "matrix": [2, 0], "x": 0.0, "y": 2.0, "h": 2 },
                { "label": "9", "matrix": [2, 1], "x": 1.0, "y": 2.0 },
                { "label": "8", "matrix": [2, 2], "x": 2.0, "y": 2.0 },
                { "label": "7", "matrix": [2, 3], "x": 3.0, "y": 2.0 },
                { "label": "6", "matrix": [3, 1], "x": 1.0, "y": 3.0 },
                { "label": "5", "matrix": [3, 2], "x": 2.0, "y": 3.0 },
                { "label": "4", "matrix": [3, 3], "x": 3.0, "y": 3.0 },
                { "label": "Enter", "matrix": [4, 0], "x": 0.0, "y": 4.0, "h": 2 },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

File: keyboards/matchstickworks/southpad/rev2/readme.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# SouthPad
+
+![Southpad](https://i.imgur.com/oXi69cu.png)
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/readme.md:3]:
```suggestion
![Southpad](https://i.imgur.com/oXi69cuh.png)
```

File: keyboards/matchstickworks/southpad/rev2/readme.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+This is a left-handed, mirrored numberpad with USB-C connection and Kailh hotswap sockets, built on the RP2040 MCU.
+
+Default layout is such:
+
+![SouthPad Layout](https://i.imgur.com/Zwxpa7a.png)
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/readme.md:9]:
```suggestion
![SouthPad Layout](https://i.imgur.com/Zwxpa7ah.png)
```

File: keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+	),
+
+	[1] = LAYOUT(
+		_______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, KC_PGUP, KC_UP,   KC_HOME,
+				 KC_LEFT, _______, KC_RGHT,
+        _______, KC_PGDN, KC_DOWN, KC_END,
+				 KC_DEL,  		   KC_INS
+	),
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c:24]:
```suggestion
    [0] = LAYOUT(
        KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
        KC_PMNS, KC_PAST, KC_PSLS, TG(1),
        KC_PPLS, KC_P9,   KC_P8,   KC_P7,
                 KC_P6,   KC_P5,   KC_P4,
        KC_PENT, KC_P3,   KC_P2,   KC_P1,
                 KC_PDOT, 		   KC_P0
	),

    [1] = LAYOUT(
        _______, _______, _______, _______,
        _______, _______, _______, _______,
        _______, KC_PGUP, KC_UP,   KC_HOME,
                 KC_LEFT, _______, KC_RGHT,
        _______, KC_PGDN, KC_DOWN, KC_END,
                 KC_DEL,  		   KC_INS
    ),
```

File: keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+	[1] = LAYOUT(
+		_______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, KC_PGUP, KC_UP,   KC_HOME,
+				 KC_LEFT, _______, KC_RGHT,
+        _______, KC_PGDN, KC_DOWN, KC_END,
+				 KC_DEL,           KC_INS
+	),
+
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c:25]:
```suggestion
    [0] = LAYOUT(
        KC_BSPC, KC_EQL,  LSFT(KC_9),LSFT(KC_0),
        KC_PMNS, KC_PAST, KC_PSLS, TG(1),
        KC_PPLS, KC_P9,   KC_P8,   KC_P7,
                 KC_P6,   KC_P5,   KC_P4,
        KC_PENT, KC_P3,   KC_P2,   KC_P1,
                 KC_PDOT, 		   KC_P0
	),

    [1] = LAYOUT(
        _______, _______, _______, _______,
        _______, _______, _______, _______,
        _______, KC_PGUP, KC_UP,   KC_HOME,
                 KC_LEFT, _______, KC_RGHT,
        _______, KC_PGDN, KC_DOWN, KC_END,
                 KC_DEL,  		   KC_INS
    ),
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/southpad/rev2/info.json [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+                { "label": "7", "matrix": [2, 3], "x": 3.0, "y": 2.0 },
+                { "label": "6", "matrix": [3, 1], "x": 1.0, "y": 3.0 },
+                { "label": "5", "matrix": [3, 2], "x": 2.0, "y": 3.0 },
+                { "label": "4", "matrix": [3, 3], "x": 3.0, "y": 3.0 },
+                { "label": "Enter", "matrix": [4, 0], "x": 0.0, "y": 4.0, "h": 2 },
+                { "label": "3", "matrix": [4, 1], "x": 1.0, "y": 4.0 },
+                { "label": "2", "matrix": [4, 2], "x": 2.0, "y": 4.0 },
+                { "label": "1", "matrix": [4, 3], "x": 3.0, "y": 4.0 },
+                { "label": ".", "matrix": [5, 1], "x": 1.0, "y": 5.0 },
+                { "label": "0", "matrix": [5, 3], "w": 2, "x": 2.0, "y": 5.0 }
```
Comment (fauxpark) [inline keyboards/matchstickworks/southpad/rev2/info.json:46]:
```suggestion
                { "label": "Bksp", "matrix": [0, 0], "x": 0, "y": 0 },
                { "label": "=", "matrix": [0, 1], "x": 1, "y": 0 },
                { "label": "(", "matrix": [0, 2], "x": 2, "y": 0 },
                { "label": ")", "matrix": [0, 3], "x": 3, "y": 0 },

                { "label": "-", "matrix": [1, 0], "x": 0, "y": 1 },
                { "label": "*", "matrix": [1, 1], "x": 1, "y": 1 },
                { "label": "/", "matrix": [1, 2], "x": 2, "y": 1 },
                { "label": "Num Lock", "matrix": [1, 3], "x": 3, "y": 1 },

                { "label": "+", "matrix": [2, 0], "x": 0, "y": 2, "h": 2 },
                { "label": "9", "matrix": [2, 1], "x": 1, "y": 2 },
                { "label": "8", "matrix": [2, 2], "x": 2, "y": 2 },
                { "label": "7", "matrix": [2, 3], "x": 3, "y": 2 },

                { "label": "6", "matrix": [3, 1], "x": 1, "y": 3 },
                { "label": "5", "matrix": [3, 2], "x": 2, "y": 3 },
                { "label": "4", "matrix": [3, 3], "x": 3, "y": 3 },

                { "label": "Enter", "matrix": [4, 0], "x": 0, "y": 4, "h": 2 },
                { "label": "3", "matrix": [4, 1], "x": 1, "y": 4 },
                { "label": "2", "matrix": [4, 2], "x": 2, "y": 4 },
                { "label": "1", "matrix": [4, 3], "x": 3, "y": 4 },

                { "label": ".", "matrix": [5, 1], "x": 1, "y": 5 },
                { "label": "0", "matrix": [5, 3], "x": 2, "y": 5, "w": 2 }
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+                 KC_PDOT, 		   KC_P0
+	),
+
+    [1] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, KC_PGUP, KC_UP,   KC_HOME,
+                 KC_LEFT, _______, KC_RGHT,
+        _______, KC_PGDN, KC_DOWN, KC_END,
+                 KC_DEL,  		   KC_INS
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/keymaps/default/keymap.c:23]:
```suggestion
                 KC_PDOT,          KC_P0
    ),

    [1] = LAYOUT(
        _______, _______, _______, _______,
        _______, _______, _______, _______,
        _______, KC_PGUP, KC_UP,   KC_HOME,
                 KC_LEFT, _______, KC_RGHT,
        _______, KC_PGDN, KC_DOWN, KC_END,
                 KC_DEL,           KC_INS
```

File: keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+                 KC_PDOT, 		   KC_P0
+	),
+
+    [1] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, KC_PGUP, KC_UP,   KC_HOME,
+                 KC_LEFT, _______, KC_RGHT,
+        _______, KC_PGDN, KC_DOWN, KC_END,
+                 KC_DEL,  		   KC_INS
```
Comment (drashna) [inline keyboards/matchstickworks/southpad/rev2/keymaps/via/keymap.c:23]:
```suggestion
                 KC_PDOT,          KC_P0
    ),

    [1] = LAYOUT(
        _______, _______, _______, _______,
        _______, _______, _______, _______,
        _______, KC_PGUP, KC_UP,   KC_HOME,
                 KC_LEFT, _______, KC_RGHT,
        _______, KC_PGDN, KC_DOWN, KC_END,
                 KC_DEL,           KC_INS
```

File: keyboards/matchstickworks/southpad/rules.mk
Comment (drashna) [inline keyboards/matchstickworks/southpad/rules.mk:1]: You either need a rules.mk that sets the default folder, or you need to add a mapping in `data/mappings/keyboard_aliases.hjson`

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matchstickworks/southpad/rules.mk
Comment (MatchstickWorks) [reply keyboards/matchstickworks/southpad/rules.mk:1]: Added mapping to keyboard_aliases

**Review by drashna — APPROVED** [review]

**Pull request merged**